src/main/java/com/radixdlt/crypto/SignaturesImpl.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.crypto;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.radixdlt.serialization.DsonOutput;
+import com.radixdlt.serialization.SerializerConstants;
+import com.radixdlt.serialization.SerializerDummy;
+
+import java.util.Map;
+
+/**
+ * A set of cryptographically secure signatures, specifying which signature was used
+ */
+public class SignaturesImpl<T extends Signature> implements Signatures {
```

Comment:
Why do we need a separate interface above this implementation? What other implementations might there be that we'll be needing soon?

Reply:
The idea is that we would not wanna "leak" implementation details to core. So `QuorumCertificate` would have an instance variable `private Signatures signatures` instead of `private SignaturesImpl<ECDSASignature> signatures`

Reply:
Also, this class should probably be package-local rather than public.

Reply:
I see, fair enough.

Reply:
@clstrfsck I removed `public`, I guess that is the access control we want?

Reply:
@clstrfsck I removed `public`, I guess that is the access control we want?

Reply:
Yeap, that's the one.